appinventor/components/src/com/google/appinventor/components/runtime/util/QUtil.java

```diff
@@ -170,6 +170,30 @@ public static String getReplAssetPath(Context context, boolean forcePrivate) {
     }
   }
 
+  /**
+   * Get the SDK-specific path where the REPL databases live. On Android Q and later, this is always
+   * an app-private directory on the "external" storage. On earlier versions of Android, this
+   * returns the external storage path for all apps (although possibly user-specific).
+   * If the {@code forcePrivate} flag is true, then the app-private directory will always be
+   * returned on devices running Android 2.2 Froyo or later.
+   *
+   * <p>
+   * For more details on why this is needed, see the deprecation notice at
+   * https://developer.android.com/reference/android/os/Environment#getExternalStorageDirectory()
+   * </p>
+   *
+   * @param context a context, such as a Form
+   * @param forcePrivate force the use of the context-specific path
+   * @return the path to the REPL's database storage
+   */
+  public static String getReplDatabasePath(Context context, boolean forcePrivate) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+      return getExternalStoragePath(context, forcePrivate) + "/databases/";
```

Comment:
I purposely placed it in the same location as the assets themselves. So if someone deletes the directory with the assets (actually its parent) the database would be deleted as well.

Reply:
I'm not sure that this is really feasible any more though since on newer devices the app-specific storage is pretty locked down. There's also the issue that in most cases it is automatically backed up if the user has opted into to backups and the database and assets might just end up being restored from the backup (but I'm less sure about this one, something to keep an eye on).

Reply:
Going to App Setting and selecting "Clear Storage" should do the trick.